packages/react-storage/src/components/StorageBrowser/context/actions/__tests__/listLocationsAction.spec.ts

```diff
@@ -1,33 +1,75 @@
-import listLocationsAction from '../listLocations';
+import {
+  createListLocationsAction,
+  ListLocationsActionInput,
+  ListLocationsOutput,
+} from '../listLocationsAction';
 
-describe('listLocationsAction', () => {
+const generateMockLocations = (size: number) => [...Array(size).keys()];
+
+const mockListLocations = jest.fn(
+  ({ pageSize }: ListLocationsActionInput): Promise<ListLocationsOutput> => {
+    // @ts-expect-error temporary badly constructed mock function
+    return Promise.resolve({
+      locations: generateMockLocations(pageSize!),
+      nextToken: undefined,
+    });
+  }
+);
+
+describe('createListLocationsAction', () => {
+  beforeEach(() => {
+    mockListLocations.mockClear();
+  });
   it('returns the expected output shape in the happy path', async () => {
+    mockListLocations.mockResolvedValueOnce({
+      // @ts-expect-error temporary workaround
+      locations: generateMockLocations(100),
+      nextToken: 'next',
+    });
+    const listLocationsAction = createListLocationsAction(mockListLocations);
+
     const output = await listLocationsAction(
-      {
-        nextToken: undefined,
-        locations: [],
-      },
-      { options: { pageSize: 100 } }
+      { nextToken: undefined, locations: [] },
+      { pageSize: 100 }
     );
 
     expect(output.locations).toHaveLength(100);
     expect(output.nextToken).toBeDefined();
   });
 
   it('merges the current action result with the previous action result', async () => {
-    const { locations, nextToken } = await listLocationsAction({
-      nextToken: undefined,
-      locations: [],
-    });
+    mockListLocations
+      .mockResolvedValueOnce({
+        // @ts-expect-error temporary workaround
+        locations: generateMockLocations(100),
+        nextToken: 'next',
+      })
+      .mockResolvedValueOnce({
+        // @ts-expect-error temporary workaround
+        locations: generateMockLocations(100),
+        nextToken: 'next-oooo',
+      });
+
+    const listLocationsAction = createListLocationsAction(mockListLocations);
+    const { locations, nextToken } = await listLocationsAction(
+      {
+        nextToken: undefined,
+        locations: [],
+      },
+      { pageSize: 100 }
+    );
 
     expect(locations).toHaveLength(100);
     expect(nextToken).toBeDefined();
 
     const { locations: nextLocations, nextToken: nextNextToken } =
-      await listLocationsAction({
-        locations,
-        nextToken,
-      });
+      await listLocationsAction(
+        {
+          locations,
+          nextToken,
+        },
+        { pageSize: 100 }
```

Comment:
Doesn't need to be in this PR, but probably move this to a const?

Reply:
For the unit tests not worried since there is no specific behavior for `pageSize` at the action level but will be adding a configuration constant for page size